03.bowling/bowling.rb

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+score = ARGV[0]
+scores = score.split(',')
+shots = []
+scores.each do |s|
+  if s == 'X'
+    shots << 10
+    shots << 0
+  else
+    shots << s.to_i
+  end
+end
+
+frames = shots.each_slice(2).to_a
+point = 0
+throw_third_last_frame = false
+
+if frames.size > 10
+  first_strike = frames[9][0] == 10
+  second_strike = frames[10][0] == 10
+
+  last_frame = if first_strike && second_strike
+                 [10, 10, frames[11][0]]
+               elsif first_strike
+                 [10, frames[10][0], frames[10][1]]
+               else
+                 [frames[9][0], frames[9][1], frames[10][0]]
+               end
+
+  frames = frames[0..8]
+  throw_third_last_frame = true
+  point += last_frame.sum
+end
+
+frames.each_with_index do |frame, i|
+  strike = frame[0] == 10
+  spare = frame.sum == 10
+  next_frame = i + 1
```

Comment:
この値はフレームのデータではなく、フレームの添え字になるので、`next_frame_index`のような名前が適切です。
ですが、そもそもの考え方として、以下のようにフレームのデータを格納した方がロジックがスッキリします。

```suggestion
  next_frame = frames[i + 1]
```